core/command.js

```diff
@@ -19,10 +19,14 @@ var $ = function(str) {
   if (!suffix) {
     return root;
   }
-  // Clear no-longer needed local variables so they don't leak into the eval.
-  str = undefined;
-  m = undefined;
-  return eval('(root)' + suffix);
+  // This regex has two main groups:
+  // 1) match .foo
+  // 2) match [42] or ['bar'] or ["baz"]
+  if (suffix.search(/^((\s*\.\s*[A-Za-z$_][A-Za-z0-9$_]*\s*)|(\s*\[\s*(\d+|\'([^\'\\]*(\\.[^\'\\]*)*)\'|"([^"\\]*(\\.[^"\\]*)*)")\s*\]\s*))+$/) === 0) {
```

Comment:
Strictly speaking this is too strict: where you have `[A-Za-z$_]` for identifier names, the spec allows *any Unicode letter*.  But AFAICT JS does not support character classes like `[[:letter:]]`, so doing better than this is not really very straight forward.